tests/integration/test_shared_db.py

```diff
@@ -2,99 +2,101 @@
-    """Test the shared-db legacy relation."""
-    charm = await ops_test.build_charm(".")
-
-    mysql_app = await ops_test.model.deploy(
-        "mysql",
-        channel="8.0/edge",
-        application_name=MYSQL_APP_NAME,
-        config={"profile": "testing"},
-        num_units=1,
-    )
-    keystone_app = await ops_test.model.deploy(
-        "keystone", application_name=KEYSTONE_APP_NAME, series="focal", num_units=2
-    )
-    # MySQLRouter is a subordinate charm, and thus needs to be deployed with no units
-    # Instead, they will be deployed with the keystone units when related with the keystone app
-    mysqlrouter_app = await ops_test.model.deploy(
-        charm,
-        application_name=MYSQLROUTER_APP_NAME,
-        num_units=None,
-        series=mysql_router_charm_series,
-    )
-
-    await ops_test.model.relate(
-        f"{KEYSTONE_APP_NAME}:shared-db", f"{MYSQLROUTER_APP_NAME}:shared-db"
-    )
-
-    async with ops_test.fast_forward():
-        await asyncio.gather(
-            ops_test.model.wait_for_idle(
-                apps=[MYSQL_APP_NAME],
-                status="active",
-                raise_on_blocked=True,
-                timeout=TIMEOUT,
-                wait_for_exact_units=1,
-            ),
-            ops_test.model.wait_for_idle(
-                apps=[KEYSTONE_APP_NAME],
-                status="blocked",
-                timeout=TIMEOUT,
-                wait_for_exact_units=2,
-            ),
-            ops_test.model.wait_for_idle(
-                apps=[MYSQLROUTER_APP_NAME],
-                status="active",
-                timeout=TIMEOUT,
-            ),
-        )
-
-    await ops_test.model.relate(
-        f"{MYSQLROUTER_APP_NAME}:backend-database", f"{MYSQL_APP_NAME}:database"
-    )
-
-    async with ops_test.fast_forward():
-        await asyncio.gather(
-            ops_test.model.block_until(lambda: mysql_app.status == "active", timeout=TIMEOUT),
-            ops_test.model.block_until(lambda: keystone_app.status == "active", timeout=TIMEOUT),
-            ops_test.model.block_until(
-                lambda: mysqlrouter_app.status == "active", timeout=TIMEOUT
-            ),
-        )
-
-    # Test that the keystone migration ran
-    get_count_keystone_tables_sql = [
-        "SELECT count(*) FROM information_schema.tables WHERE table_schema = 'keystone'",
-    ]
-
-    mysql_unit = ops_test.model.applications[MYSQL_APP_NAME].units[0]
-    mysql_unit_address = await mysql_unit.get_public_address()
-
-    server_config_credentials = await get_server_config_credentials(mysql_unit)
-
-    output = await execute_queries_on_unit(
-        mysql_unit_address,
-        server_config_credentials["username"],
-        server_config_credentials["password"],
-        get_count_keystone_tables_sql,
-    )
-    assert output[0] > 0
+# TODO: re-enable when bug resolved: https://bugs.launchpad.net/charm-keystone/+bug/1990243
```

Comment:
i am unsure if commenting these tests is ideal - i would rather they be deleted if they are not running or mark them unstable. is there any reason they're being commented instead?

Reply:
I would be in favor of deleting them. Thoughts @paulomach?

Reply:
Q: https://bugs.launchpad.net/charm-keystone/+bug/1990243 is marked as "Will not fix" (as already fixed in another library). If the test is still broken, we need to reopen the ticket, otherwise re-enable the test. Isn't it?